### PR TITLE
[#457] Improve institutions endpoint

### LINKED ISSUE #457: Improve institutions endpoint to work for any signed in user
Right now `/api/v1/institutions` throws `403` for any user who is not a superuser.
This is not desirable, as a group manager who has access to to multiple institutions can not list them. It is possible to retrive slugs to the institutions from user permissions, but not institution's name, which is necessary to present a human readable modal and UI.

Desired solution:

Any signed in user should be able to access `/api/v1/institutions` but the list should be filtered based on what the user has access to.

### PR DESCRIPTION
**Pull request by kmarszalek**
- return all institutions where user has any permissions

Closes #457.

### CODE REVIEW

**Comment by kmarszalek** [comment]
@wziajka https://pr-464-sok.docker-fid.grid.cyf-kr.edu.pl/ 
code: 503

File: s4e-backend/src/test/java/pl/cyfronet/s4e/controller/InstitutionControllerTest.java [hunk 1]
```diff
@@ -0,0 +1,172 @@
...
+
+    @Autowired
+    private SlugService slugService;
+
+    @Autowired
+    private MockMvc mockMvc;
+
+    private AppUser appUser;
+
+    private String test_institution = "Test Institution";
```
Comment (jswk) [inline s4e-backend/src/test/java/pl/cyfronet/s4e/controller/InstitutionControllerTest.java:66]: Wouldn't it be better to use camelCase here?

File: s4e-backend/src/test/java/pl/cyfronet/s4e/controller/InstitutionControllerTest.java [hunk 2]
```diff
@@ -0,0 +1,172 @@
...
+        mockMvc.perform(get(API_PREFIX_V1 + "/institutions")
+                .contentType(MediaType.APPLICATION_JSON)
+                .with(jwtBearerToken(appUser, objectMapper)))
+                .andExpect(status().isOk())
+                .andExpect(jsonPath("$.length()", is(equalTo(0))));
+    }
+
+    @Test
+    public void shouldGetAllUsersInstitutions() throws Exception {
+        String test_institution = "Test Institution ZK";
```
Comment (jswk) [inline s4e-backend/src/test/java/pl/cyfronet/s4e/controller/InstitutionControllerTest.java:125]: Why not reuse the institution from @BeforeEach?

Comment (kmarszalek) [reply s4e-backend/src/test/java/pl/cyfronet/s4e/controller/InstitutionControllerTest.java]: beforeeach creates institution for each test, here i need two more.

File: s4e-backend/src/test/java/pl/cyfronet/s4e/controller/InstitutionControllerTest.java [hunk 3]
```diff
@@ -0,0 +1,172 @@
...
+                .contentType(MediaType.APPLICATION_JSON)
+                .with(jwtBearerToken(appUser, objectMapper)))
+                .andExpect(status().isOk())
+                .andExpect(jsonPath("$.length()", is(equalTo(0))));
+    }
+
+    @Test
+    public void shouldGetAllUsersInstitutions() throws Exception {
+        String testInstitution = "Test Institution ZK";
+        String slugInstitution = slugService.slugify(testInstitution);
```
Comment (jswk) [inline s4e-backend/src/test/java/pl/cyfronet/s4e/controller/InstitutionControllerTest.java:126]: I'd refactor the names of these vars not to override class fields.

File: s4e-backend/src/test/java/pl/cyfronet/s4e/controller/InstitutionControllerTest.java [hunk 4]
```diff
@@ -0,0 +1,172 @@
...
+                .name(testInstitution)
+                .slug(slugInstitution)
+                .build());
+        Group group = groupRepository.save(Group.builder().name("__default__").slug("default").institution(institution).build());
+
+        UserRole userRole = UserRole.builder().role(AppRole.GROUP_MEMBER).user(appUser).group(group).build();
+        userRoleRepository.save(userRole);
+
+        testInstitution = "Test Institution PAK";
+        slugInstitution = slugService.slugify(testInstitution);
```
Comment (jswk) [inline s4e-backend/src/test/java/pl/cyfronet/s4e/controller/InstitutionControllerTest.java:137]: And not redefine them here as well.
Use names such as `testInstitutionZK`, `slugInstZK`, etc.

**Pull request merged**